src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPoolWorkQueue.AutoreleasePool.OSX.cs

```diff
@@ -0,0 +1,38 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.CompilerServices;
+using System.Runtime.Versioning;
+
+namespace System.Threading
+{
+    internal sealed partial class ThreadPoolWorkQueue
+    {
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        [SupportedOSPlatform("macos")]
+        [SupportedOSPlatform("ios")]
+        [SupportedOSPlatform("tvos")]
+        [SupportedOSPlatform("watchos")]
+        private static void DispatchItemWithAutoreleasePool(object workItem, Thread currentThread)
+        {
+            IntPtr autoreleasePool = Interop.Sys.CreateAutoreleasePool();
+            try
+            {
+#pragma warning disable CS0162 // Unreachable code detected. EnableWorkerTracking may be a constant in some runtimes.
+                if (ThreadPool.EnableWorkerTracking)
```

Comment:
This is missing `PortableThreadPoolEventSource.Log.IsEnabled()` condition or it will increase the size unnecessary. Alternatively, the better fix could be to change EnableWorkerTracking to static property and substitute it under existing `System.Diagnostics.Tracing.EventSource.IsSupported`

@eerhardt

Reply:
I matched the same condition as the other place we check `EnableWorkerTracking`.

Reply:
I think that one should be changed as well

Reply:
Can we do that in a separate PR? I want to keep this PR focused on the AutoreleasePool support.

Reply:
Yep if you don't forget about it ;-)

Reply:
Opened https://github.com/dotnet/runtime/issues/47871 to track the work in case I don't get to it right away.